### DISCUSSION TITLE
any plan to supporting VIEW or Materialized VIEW in ArcadeDB SQL?

### DISCUSSION THREAD

**Discussion by frank-dkvan** [ArcadeData/arcadedb#458] [Q&A] [answered]
Hi @lvca , I remember OrientDB has supported view and materialized view already?  is it easy to support  view and materialized view  features in ArcadeDB SQL?

**Comment by lvca** [comment, marked as answer]
@frank-dkvan it's not on our roadmap. You could obtain the a similar feature by running the queries with a scheduler and storing the records flat on some bucket. Then drop the buckets and create them from scratch every time. This would be a materialized view with a little coding.

**Reply by frank-dkvan** [reply]
Hi @lvca , is there document about SQL extension development?  maybe I can develop the features later on ....saying:  CREATE MATERIALIZED VIEW MV_NAME_XXX AS  ..............

**Reply by lvca** [reply]
ArcadeDB SQL engine has been ported from OrientDB. So if you can get the missing pieces (syntax and classes) from OrientDB, it shouldn't be very hard.

Check these resources:

- OrientDB SQL syntax: https://github.com/orientechnologies/orientdb/blob/develop/core/src/main/grammar/OrientSQL.jjt
- OrientDB Create View Statement: https://github.com/orientechnologies/orientdb/blob/6e94047200f6b7a15c02902eea21a26a85351fe6/core/src/main/java/com/orientechnologies/orient/core/sql/parser/OCreateViewStatement.java
